node/node.go

```diff
@@ -256,7 +255,7 @@ func newTxValidator(pool mempool.Mempool, poolIDs *mempoolIDs, logger log.Logger
 	}
 }
 
-func newHeaderValidator(logger log.Logger) p2p.GossipValidator {
+func newHeaderValidator(logger log.Logger, incomingHeaderCh chan *p2p.GossipMessage) p2p.GossipValidator {
```

Comment:
Consider changing this function to a method of type `Node` (after adding receiver you can omit both arguments).